Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp

```diff
@@ -152,18 +152,38 @@ class InternalSource final : public MediaStreamTrackPrivate::Observer,
         }), this);
 
 #if USE(GSTREAMER_WEBRTC)
-        auto pad = adoptGRef(gst_element_get_static_pad(m_src.get(), "src"));
-        gst_pad_add_probe(pad.get(), GST_PAD_PROBE_TYPE_EVENT_UPSTREAM, reinterpret_cast<GstPadProbeCallback>(+[](GstPad*, GstPadProbeInfo* info, InternalSource* internalSource) -> GstPadProbeReturn {
-            auto& trackSource = internalSource->m_track.source();
+        auto& trackSource = m_track.source();
+        if (trackSource.isIncomingAudioSource() || trackSource.isIncomingVideoSource()) {
+
             if (trackSource.isIncomingAudioSource()) {
                 auto& source = static_cast<RealtimeIncomingAudioSourceGStreamer&>(trackSource);
-                source.handleUpstreamEvent(GRefPtr<GstEvent>(GST_PAD_PROBE_INFO_EVENT(info)));
+                m_webrtcSourceClientId = source.registerClient(GRefPtr<GstElement>(m_src));
             } else if (trackSource.isIncomingVideoSource()) {
                 auto& source = static_cast<RealtimeIncomingVideoSourceGStreamer&>(trackSource);
-                source.handleUpstreamEvent(GRefPtr<GstEvent>(GST_PAD_PROBE_INFO_EVENT(info)));
+                m_webrtcSourceClientId = source.registerClient(GRefPtr<GstElement>(m_src));
             }
-            return GST_PAD_PROBE_OK;
-        }), this, nullptr);
+
+            auto pad = adoptGRef(gst_element_get_static_pad(m_src.get(), "src"));
+            gst_pad_add_probe(pad.get(), static_cast<GstPadProbeType>(GST_PAD_PROBE_TYPE_EVENT_UPSTREAM | GST_PAD_PROBE_TYPE_QUERY_UPSTREAM), reinterpret_cast<GstPadProbeCallback>(+[](GstPad*, GstPadProbeInfo* info, InternalSource* internalSource) -> GstPadProbeReturn {
+                auto& trackSource = internalSource->m_track.source();
+                auto clientId = internalSource->m_webrtcSourceClientId.value();
+                ASSERT(clientId);
```

Comment:
Shouldn't you assert if that has a value and then then it?

Reply:
can't parse the question :)

Reply:
Do you mean
```
ASSERT(internalSource->m_webrtcSourceClientId)
auto clientId = internalSource->m_webrtcSourceClientId.value();
```

¿

Reply:
Sorry, I meant that I think you should assert on the value to have an actual value and then retrieving it. Now you're doing the other way around and it looks weird, but I might be missing something.